### PR TITLE
Joke challenge part 2

### PR DESCRIPTION
**Pull request by Vilius-kul**

### CODE REVIEW

File: schemas.py [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
         return value
+
+    #language selection validation limited to 4 languages
+    @validator('language')
+    def language_validator(cls, value):
+        language_selection = ['pl','lt','da','ru']
+        if value.lower() not in language_selection:
+            raise ValueError("Wrong input! Please type 'pl' for Polish,\
+                            'lt' for Lithuanian, 'da' for Danish or 'ru' for Russian")
+        return value.lower()
```
Comment (SF92) [inline schemas.py:21]: Very nice

File: translate.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+
+class JokeTranslator:
+
+    #mock translator
+    @staticmethod
+    def get_joke(jokes, langInput):
```
Comment (SF92) [inline translate.py]: I'd probable make this more generic for flexibility. Something like:

```
class Translator:

    @staticmethod
    def translate(text: str, language: str) -> str:
          return "translated text"
```

Then later when using this class you can do something like:

```
translated_jokes = [Translator.translate(joke) for joke in jokes]
```

File: random_joke.py [hunk 3]
```diff
@@ -23,9 +23,23 @@ def multiple_joke():
         request_data = MultipleJokesRequestParams(**raw_request)
     except ValidationError as exc:
         return(str(exc))
-        
+
     return JokeApi.multiple_jokes(request_data.count)
 
+# mock/test translator
+@app.route("/multi-language-jokes")
+def test_translator():
```
Comment (SF92) [inline random_joke.py]: Is this supposed to be called `test_translator`?

File: random_joke.py [hunk 4]
```diff
@@ -23,9 +23,23 @@ def multiple_joke():
...
+    # Raw query data
+    raw_request = request.args
+    try:
+        request_data = MultipleJokesRequestParams(**raw_request)
+    except ValidationError as exc:
+        return(str(exc))
+    joke = JokeApi.multiple_jokes(request_data.count)
```
Comment (SF92) [inline random_joke.py]: This variable should probably be named `jokes` for clarity

**Review by SF92 — COMMENTED** [review]
Good work, a little bit of cleanup suggested before moving on to the actual api integration.

Regarding api keys, yes the best way is to have them as environment variables. For now the simplest way to implement that would be to have a separate `settings.py` file that looks lke:

```
import os

TRANSLATION_API_KEY = os.environ.get("TRANSLATION_API_KEY", "")
```

then you can import `from src.settings import TRANSLATION_API_KEY` as needed.

To set the env var just run:

`export TRANSLATION_API_KEY=49ur4h43hjf49fe` in the terminal before running your regular flask command

File: random_joke.py (hunk 3 above)
Comment (Vilius-kul) [reply random_joke.py]: No, It all mine :D

File: schemas.py (hunk 1 above)
Comment (Vilius-kul) [reply schemas.py:21]: Thank you

File: translate.py (hunk 2 above)
Comment (Vilius-kul) [reply translate.py]: I have done it a little bit different, let me know if that's ok.

File: random_joke.py (hunk 3 above)
Comment (Vilius-kul) [reply random_joke.py]: renamed

**Review by SF92 — APPROVED** [review]

File: translate.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        #from dict to str + new line after each joke
+        string =''
+        for j in translation['translations']:
+            string += j['translation']+"<br>"
+        return string
+
+    #mock translator
+    @staticmethod
+    def translate(text: str, language: str) -> str:
+        jstring ='<br>'.join([j for j in text])
```
Comment (SF92) [inline translate.py]: I would recommend not injecting html markup into the response as there is no guarantee tha the client using this api will want the response in an html format. We can keep as is for now but later down the line you will likely see the need to change it.

**Pull request merged**